test/test-framework-spec.md

```diff
@@ -143,53 +189,55 @@ The following are the list of available assertions available in the test framewo
 
 <table>
   <tr>
-   <td>@test:assertEquals
+   <td>@test:assertEquals(any|error actual, anydata expected, string message)
    </td>
-   <td>Checks if the specified value is equal to the expected value.
+   <td>Checks if the specified value is equal to the expected value. This assertion relies on `==` based deep equality check. 
+For the deep equality check to be applicable, it is required that at least one of the participating operands is of a static type that is a subtype of `anydata`. 
+Consequently, the permissible scope of the expected value is confined to `anydata` type.
    </td>
   </tr>
   <tr>
-   <td>@test:assertNotEquals
+   <td>@test:assertNotEquals(any actual, anydata expected, string message)
    </td>
-   <td>Checks if the specified value is not equal to the expected value.
+   <td>Checks if the specified value is not equal to the expected value. This assertion utilizes the != operator for a deep equality check, thus limiting the expected value to the `anydata` type.
+The actual value does not include `error` type to avoid automatically passing the tests when there is an error.
+This ensures that tests accurately reflect intended outcomes without misinterpreting errors as correct behavior.
    </td>
   </tr>
   <tr>
-     <td>@test:assertExactEquals
+     <td>@test:assertExactEquals(any|error actual, any|error expected, string message)
      </td>
-     <td>Checks if the specified value is exactly equal to the expected value
-     i.e. both refer to the same entity.
+     <td>Checks if the specified value is exactly equal to the expected value i.e. both refer to the same entity. This assertion utilizes the === operator for an exact equality check.
      </td>
     </tr>
     <tr>
      <td>@test:assertNotExactEquals
      </td>
-     <td>Checks if the specified value is not exactly equal to the expected value
-     i.e. both do not refer to the same entity.
+     <td>Checks if the specified value is not exactly equal to the expected value i.e. both do not refer to the same entity.
+This assertion utilizes the === operator for an exact equality check.
      </td>
     </tr>
   <tr>
-   <td>@test:assertTrue
+   <td>@test:assertTrue(boolean expression, string message)	
    </td>
-   <td>Checks if the specified value is true.
+   <td>Checks if the specified boolean expression is true.
    </td>
   </tr>
   <tr>
    <td>@test:assertFalse
    </td>
-   <td>Checks if the specified value is false.
+   <td>Checks if the specified boolean expression is false.
    </td>
   </tr>
   <tr>
    <td>@test:assertFail
    </td>
-   <td>Forces a test case to fail.
+   <td>Forces a test case to fail. This assertion facilitate the failing of a test during its execution when a specific condition is not met.
```

Comment:
"This assertion facilitate the failing of a test during its execution when a specific condition is not met." - is this correct? Other assert functions also will fail the test if a specific condition is not met.

Reply:
Agreed that it's confusing. I'll fix it.